core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresQueryDao.scala

```diff
@@ -430,28 +412,38 @@ private[r2dbc] class PostgresQueryDao(settings: R2dbcSettings, executorProvider:
   }
 
   override def persistenceIds(entityType: String, afterId: Option[String], limit: Long): Source[String, NotUsed] = {
+    val actualLimit = if (limit > Int.MaxValue) Int.MaxValue else limit.toInt
     val likeStmtPostfix = PersistenceId.DefaultSeparator + "%"
-    // FIXME
-    require(settings.numberOfDataPartitions == 1, "persistenceIds not implemented for more than one data-partition yet")
-    val executor = executorProvider.executorFor(slice = 0)
-    val result = executor.select(s"select persistenceIds by entity type")(
-      connection =>
-        afterId match {
-          case Some(after) =>
-            val stmt = connection.createStatement(persistenceIdsForEntityTypeAfterSql)
-            bindPersistenceIdsForEntityTypeAfterSql(stmt, entityType, likeStmtPostfix, after, limit)
-
-          case None =>
-            val stmt = connection.createStatement(persistenceIdsForEntityTypeSql)
-            bindPersistenceIdsForEntityTypeSql(stmt, entityType, likeStmtPostfix, limit)
+    val results: immutable.IndexedSeq[Future[immutable.IndexedSeq[String]]] =
```

Comment:
Would it be better to query one partition at a time and stream out those results than to load all results from all partitions? Maybe fine since only persistence ids?

Reply:
I think the result must be sorted, for the paged queries, and that is used also by the ordinary `currentPersistenceIds` which is using a ContinuousQuery to fetch all in chunks.

Reply:
Didn't think of the ordering, yes, I can't see any other way than this given that.

Reply:
Actually, for the `currentPersistenceIds()` which retrieves all in chunks, it's probably possible to concat several of those queries where each go to a separate data partition. Would result in a different total order than we have today, but that might be fine. However, the other queries with user defined paging key can't use that, and therefore I prefer to keep them the same.